ee/connectors/utils/project_utils.py

```diff
@@ -0,0 +1,95 @@
+from utils.pg_client import PostgresClient
+from queue import Queue
+from decouple import config
+from time import time
+
+
+def _project_from_session(sessionId):
+    with PostgresClient() as conn:
+         conn.execute(
+                 conn.mogrify("SELECT project_id FROM sessions WHERE session_id=%(sessionId)s LIMIT 1",
+                             {'sessionId': sessionId})
+                 )
+         res = conn.fetchone()
+    if res is None:
+        print('[WARN] Session not found in sessions ids')
+        return None
+    return res['project_id']
+
+
+class SessionsHistory:
+
+    def __init__(self):
+        self.session_project = dict()
+        self.max_alive_time = config('MAX_SESSION_LIFE', default=7200) # Default 2 hours
+
+    def add_new(self, sessionid):
+        self.session_project[sessionid] = (time(), 'OPEN')
+
+    def add(self, sessionid):
+        if sessionid in self.session_project.keys():
+            if self.session_project[sessionid][1] == 'CLOSE':
+                tmp = self.session_project[sessionid]
+                self.session_project[sessionid] = (tmp[0], 'UPDATE')
+        else:
+            self.add_new(sessionid)
+
+    def close(self, sessionid):
+        tmp = self.session_project[sessionid]
+        old_status = tmp[1]
+        self.session_project[sessionid] = (tmp[0], 'CLOSE')
+        return old_status
+
+    def clear_sessions(self):
+        to_clean_list = list()
+        current_time = time()
+        for sessionid, values in self.session_project.items():
+            if current_time - values[0] > self.max_alive_time:
+                to_clean_list.append(sessionid)
+                del self.session_project[sessionid]
+        return to_clean_list
+
+
+class ProjectSelection:
+
+    def __init__(self, selection=list()):
+        self.selection = selection
+        self.cache_true = list()
```

Comment:
SessionStructure{
  toFilter bool
  lastUpdate timestamp
}

CleanOldSessions() {
  now = time.Now() // current timestamp
  for session in self.sessions {
    if now - session.lastUpdate > 15 minutes {
      // delete session from cache
    }
  }
}